project/apps/api-gateway/src/user/user.controller.ts

```diff
@@ -0,0 +1,132 @@
+  
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  HttpStatus,
+  Param,
+  Patch,
+  Put,
+  Req,
+  Res,
+  Inject,
+} from '@nestjs/common';
+import { ClientProxy } from '@nestjs/microservices';
+import {
+  UpdateUserDto,
+  updateUserSchema
+} from '@repo/dtos/user';
+import { ZodValidationPipe } from '@repo/pipes/zod-validation-pipe.pipe';
+import { Request, Response } from 'express';
+import { firstValueFrom } from 'rxjs';
+@Controller('user')
+export class UserController {
+  constructor(
+    @Inject('USER_SERVICE')
+    private readonly authServiceClient: ClientProxy,
+    private readonly userServiceClient: ClientProxy,
```

Comment:
i think the userServiceClient hasnt been registered yet, need to register it in `project/apps/user-service/src/user.module.ts`